File: C++/isomorphic_strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int key[128] = {0}, value[128] = {0};
        
        for(int i = 0; i < s.size(); i++) {
            if(key[s[i]] != value[t[i]]) return false;
            key[s[i]] = i + 1;
            value[t[i]] = i + 1;
        }
        return true;
    }
};

File: Python3/isomorphic_strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashmap = {}
        
        for i in range(len(s)):
            if s[i] in hashmap and hashmap[s[i]] != t[i]:
                return False
            elif s[i] not in hashmap and t[i] in hashmap.values():
                return False
            else:
                hashmap[s[i]] = t[i]
        
        return True
